src/libraries/System.Net.Mail/src/System/Net/Mail/MailAddress.cs

```diff
@@ -63,6 +63,61 @@ public MailAddress(string address, string displayName) : this(address, displayNa
         //
         // A FormatException will be thrown if any of the components in 'address' are invalid.
         public MailAddress(string address, string displayName, Encoding displayNameEncoding)
+        {
+            bool parseSuccess = TryParse(address, displayName, displayNameEncoding,
+                                        out (string displayName, string user, string host, Encoding displayNameEncoding) parsedData,
+                                        throwExceptionIfFail: true);
+
+            _displayName = parsedData.displayName;
+            _userName = parsedData.user;
+            _host = parsedData.host;
+            _displayNameEncoding = parsedData.displayNameEncoding;
+
+            Debug.Assert(parseSuccess);
+        }
+
+        /// <summary>
+        /// Create a new <see cref="MailAddress"/>. Does not throw an exception if the MailAddress cannot be created.
+        /// </summary>
+        /// <param name="address">A <see cref="string"/> that contains an email address.</param>
+        /// <param name="result"></param>
+        /// <returns>A <see cref="bool"/> value that is true if the <see cref="MailAddress"/> was successfully created; otherwise, false.</returns>
+        public static bool TryCreate(string address, out MailAddress result) => TryCreate(address, displayName: null, out result);
+
+        /// <summary>
+        /// Create a new <see cref="MailAddress"/>. Does not throw an exception if the MailAddress cannot be created.
+        /// </summary>
+        /// <param name="address">A <see cref="string"/> that contains an email address.</param>
+        /// <param name="displayName">A <see cref="string"/> that contains the display name associated with address. This parameter can be null.</param>
+        /// <param name="result"></param>
```

Comment:
Empty param

Reply:
👍

Reply:
Updated here and on other...I'm not so great with documentation, let me know and feel free to suggest something better!I took inspiration from int.TryParse.